lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -53,6 +53,17 @@ def available_replicas
 
     private
 
+    def stale_status?
+      observed_generation != current_generation
+    end
+
+    def check_pods?
```

Comment:
`check_pods_for_fast_failure?`

Reply:
I still feel like this method is a bit "dense" with context and ideas that I will immediately forget and then look at this again in 2 weeks and have no idea what each of those lines means. Do you think that's only because I'm not familiar enough with this project or do you think others might have the same problem? If so, maybe clarifying with comments a bit might help?
```ruby
def check_pods?
  # No point checking for pods if the replica set itself doesn't exist
  return false unless exists?

  # dunno I don't really understand this. I guess don't check pods if the replica set we have is already stale?
  return false if stale_status?

  # if we don't care about how many pods succeed, we don't need to look for them
  return false if desired_replicas == 0

  # 2 instead of 1 because of some magical safe-guard for some edge case. imho worth explaining so that
  # the context isn't lost
  ready_replicas < 2
end
```

Reply:
I think anyone might be confused. I added more comments and changed the method to `fetch_pods_if_necessary` which I think makes this easier to follow overall.